### PR TITLE
Fix for ONLY_POST_ALLOWED | Change HTTP methods from GET to POST due to public IP change

### PR DESCRIPTION
**Pull request by IutkinEgor**
Not a Go developer, but it works.

P.s.

You don't have to wait for the merge request; you can already pull the built image from Docker Hub for this fork:

```
docker pull iutkined/cluster-issuer-regru:1.1.2
```

and replace in `values.yaml` with: 

```yaml 
issuer:
  image: iutkined/cluster-issuer-regru:1.1.2
```

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -27,73 +28,74 @@ func NewRegruClient(username string, password string, zone string) *RegruClient
...
+	writer.WriteField("password", c.password)
+	writer.Close()
+
+	// Perform the POST request
+	req, err := http.NewRequest("POST", apiURL, &b)
 	if err != nil {
-		return fmt.Errorf("failed to make GET request: %v", err)
+		return fmt.Errorf("failed to create request: %v", err)
 	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
```
Comment (kneumoin) [inline client.go:66]: Why need to use writer.FormDataContentType() instead of Content-Type: application/json? Does the API require multipart/form-data?

**Comment by c0rnoTa** [comment]
Fix same issue https://github.com/c0rnoTa/cert-manager-webhook-regru/commit/c780607ea480fd2b9591974faf60353383624f17

You are welcome to use `ghcr.io/c0rnota/cluster-issuer-regru:1.1.6`

**Review by unreturned — APPROVED** [review]

**Review by kneumoin — COMMENTED** [review]
ack

**Review by kneumoin — APPROVED** [review]

File: client.go (hunk 1 above)
Comment (unreturned) [reply client.go:66]: Yes, API require multipart/form-data

Comment (unreturned) [reply client.go:66]:
```
2.2.2. Передача входных параметров
Все дополнительные параметры, если они есть, можно передавать в виде стандартных HTTP-параметров GET или POST, где передаваемые данные кодируются как x-www-form-urlencoded.

Пример передачи параметров через GET запрос:

https://api.reg.ru/api/regru2/nop?username=test&password=test&output_content_type=plain

При использовании curl необходимо URL с параметрами заключать в кавычки во избежание интерпретации shell-интерпретатором спецсимволов в URL.
```

https://www.reg.ru/reseller/api2doc

**Pull request merged**